src/sentry/identity/oauth2.py

```diff
@@ -30,24 +31,88 @@ class OAuth2Provider(Provider):
 
     oauth_scopes = ()
 
+    def get_oauth_access_token_url(self):
+        # for most cloud-based providers, take this off of the class
+
+        if self.oauth_access_token_url is not '':
+            return self.oauth_access_token_url
+
+        # # check the model for instalation information
+        # if self.provider_model.config and self.provider_model.config.get('XXXXX'):
+        #     # todo(maxbittker) get the name of this key
+        #     return self.provider_model.config.get('XXXXX')
+
+        # otherwise try the pipeline state
+        pipeline_token_url = self.pipeline.parent_pipeline.fetch_state(
+            'oauth_config_information').get('access_token_url')
+
+        if pipeline_token_url:
+            return pipeline_token_url
+
+        raise NotImplementedError
+
+    def get_oauth_authorize_url(self):
+        # for most cloud-based providers, take this off of the class
+
+        if self.oauth_authorize_url is not '':
+            return self.oauth_authorize_url
+
+        # # check the model for instalation information
+        # if self.provider_model.config and self.provider_model.config.get('XXXXX'):
+        #     # todo(maxbittker) get the name of this key
+        #     return self.provider_model.config.get('XXXXX')
+
+        # otherwise try the pipeline state
+
+        pipeline_authorize_url = self.pipeline.parent_pipeline.fetch_state(
+            'oauth_config_information').get('authorize_url')
+
+        if pipeline_authorize_url:
+            return pipeline_authorize_url
+
+        raise NotImplementedError
+
     def get_oauth_client_id(self):
+        # # check the model for instalation information
+        # if self.provider_model.config and self.provider_model.config.get('XXXXX'):
+        #     # todo(maxbittker) get the name of this key
+        #     return self.provider_model.config.get('XXXXX')
+
+        # otherwise try the pipeline state
+        client_id = self.pipeline.parent_pipeline.fetch_state('oauth_config_information').get('id')
+
+        if client_id:
+            return client_id
+
         raise NotImplementedError
 
     def get_oauth_client_secret(self):
+        # # check the model for instalation information
+        # if self.provider_model.config and self.provider_model.config.get('XXXXX'):
+        #     # todo(maxbittker) get the name of this key
+        #     return self.provider_model.config.get('XXXXX')
+
+        # otherwise try the pipeline state
+        client_secret = self.pipeline.parent_pipeline.fetch_state('oauth_config_information') \
+            .get('secret')
+
+        if client_secret:
+            return client_secret
+
         raise NotImplementedError
 
     def get_oauth_scopes(self):
         return self.config.get('oauth_scopes', self.oauth_scopes)
 
     def get_pipeline_views(self):
         return [
-            OAuth2LoginView(
-                authorize_url=self.oauth_authorize_url,
+            lambda: OAuth2LoginView(
```

Comment:
this was needed to be postponed because we can't evaluate the urls until earlier parts of the flow have completed